dataherald/api/fastapi.py

```diff
@@ -221,10 +222,36 @@ def list_table_descriptions(
         self, db_connection_id: str | None = None, table_name: str | None = None
     ) -> list[TableSchemaDetail]:
         scanner_repository = DBScannerRepository(self.storage)
-        return scanner_repository.find_by(
+        table_descriptions = scanner_repository.find_by(
             {"db_connection_id": db_connection_id, "table_name": table_name}
         )
 
+        if db_connection_id:
+            db_connection_repository = DatabaseConnectionRepository(self.storage)
+            db_connection = db_connection_repository.find_by_id(db_connection_id)
+            database = SQLDatabase.get_sql_engine(db_connection)
+            inspector = inspect(database.engine)
+            meta = MetaData(bind=database.engine)
+            MetaData.reflect(meta, views=True)
+            all_tables = inspector.get_table_names() + inspector.get_view_names()
+
+            for table_description in table_descriptions:
+                if table_description.table_name not in all_tables:
+                    table_description.status = "DEPRECATED"
+                else:
+                    all_tables.remove(table_description.table_name)
```

Comment:
@jcjc712 what is DEPRECATED?

Reply:
- `DEPRECATED` if there is a row in our `table-descriptions` collection that is no longer in the database, probably because the table/view was deleted or renamed